internal/server/stat_point.go

```diff
@@ -247,12 +379,148 @@ func (s *Server) GetStatSetWithinPlace(
+					result.Data[statVar].Metadata[metaHash] = meta
 				}
 			}
 		}
 	}
 
 	return result, nil
 }
+
+// GetStatSetWithinPlaceAll implements API for Mixer.GetStatSetWithinPlaceAll.
+// Endpoint: /stat/set/within-place/all
+func (s *Server) GetStatSetWithinPlaceAll(
+	ctx context.Context, in *pb.GetStatSetWithinPlaceRequest) (
+	*pb.GetStatSetAllResponse, error,
+) {
+	parentPlace := in.GetParentPlace()
+	statVars := in.GetStatVars()
+	childType := in.GetChildType()
+	date := in.GetDate()
+	log.Printf(
+		"GetStatSetWithinPlaceAll: parentPlace: %s, statVars: %v, childType: %s, date: %s",
+		parentPlace,
+		statVars,
+		childType,
+		date,
+	)
+	if parentPlace == "" {
+		return nil, status.Errorf(codes.InvalidArgument,
+			"Missing required argument: parent_place")
+	}
+	if len(statVars) == 0 {
+		return nil, status.Errorf(codes.InvalidArgument,
+			"Missing required argument: stat_vars")
+	}
+	if childType == "" {
+		return nil, status.Errorf(codes.InvalidArgument,
+			"Missing required argument: child_type")
+	}
+	dateKey := date
+	if date == "" {
+		dateKey = "LATEST"
+	}
+
+	hs := &util.HashStore{}
+	ctx = context.WithValue(ctx, util.HashStoreKey, hs)
+
+	// Pre-populate result
+	result := &pb.GetStatSetAllResponse{
+		Data: make(map[string]*pb.PlacePointStatAll),
+	}
+	for _, statVar := range statVars {
+		result.Data[statVar] = &pb.PlacePointStatAll{
+			StatList: []*pb.PlacePointStat{},
+		}
+	}
+
+	// Read from cache directly
+	rowList, keyTokens := buildStatSetWithinPlaceKey(parentPlace, childType, dateKey, statVars)
+	cacheData, err := readStatCollection(ctx, s.store, rowList, keyTokens)
+	if err != nil {
+		return nil, err
+	}
+
+	gotResult := false
+	for _, statVar := range statVars {
+		data, ok := cacheData[statVar]
+		if !ok || data == nil {
+			continue
+		}
+		gotResult = true
+		for _, cohort := range data.SourceCohorts {
+			// The cohort is from the same source.
+			metaData := &pb.StatMetadata{
+				MeasurementMethod: cohort.MeasurementMethod,
+				ObservationPeriod: cohort.ObservationPeriod,
+				ProvenanceUrl:     cohort.ProvenanceUrl,
+				ScalingFactor:     cohort.ScalingFactor,
+				ImportName:        cohort.ImportName,
+				Unit:              cohort.Unit,
+			}
+			metaHash := hs.GetHash(metaData)
+			pointStat := &pb.PlacePointStat{
+				Stat:     map[string]*pb.PointStat{},
+				Metadata: map[string]*pb.StatMetadata{metaHash: metaData},
+			}
+			for place, val := range cohort.Val {
+				pointStat.Stat[place] = &pb.PointStat{
+					Date:     cohort.PlaceToLatestDate[place],
+					Value:    val,
+					MetaHash: metaHash,
+				}
+			}
+			result.Data[statVar].StatList = append(result.Data[statVar].StatList, pointStat)
+		}
+	}
+	// Check if need to read from memory database.
+	statVarInMemDb := false
+	for _, statVar := range statVars {
+		if s.store.MemDb.HasStatVar(statVar) {
+			statVarInMemDb = true
+			break
+		}
+	}
+	// Need to fetch child places if to read data from memory database or
```

Comment:
nit: "...if to read data..." -> "...if need to read data ..."

Reply:
Done